lib/hpcbase.pm

```diff
@@ -219,4 +219,40 @@ sub prepare_user_and_group {
     assert_script_run('useradd -u 7777 -g 7777 slurm');
 }
 
+=head2 prepare_spack_env
+
+  prepare_spack_env($mpi)
+
+After install spack and HPC C<mpi> required packages, prepares env
+variables. The HPC packages (*-gnu-hpc) use an installation path that is separate
+from the rest and can be exported via a network file system.
+
+After C<prepare_spack_env> run, C<spack> should be ready to build entire tool stack,
+downloading and installing all bits required for whatever package or compiler.
+=cut
+sub prepare_spack_env {
+    my ($self, $mpi) = @_;
+    $mpi //= 'mpich';
+    zypper_call "in spack $mpi-gnu-hpc $mpi-gnu-hpc-devel";
+    $self->relogin_root;
+    assert_script_run 'source /etc/profile.d/lmod.sh';
```

Comment:
After a relogin you should not have to source /etc/profile.d/lmod.sh. If you do, something would be wrong.

Reply:
should this be before relogin?